fs/copy.go

```diff
@@ -32,14 +33,52 @@ var bufferPool = &sync.Pool{
 	},
 }
 
+// XAttrErrorHandlers transform a non-nil xattr error.
+// Return nil to ignore an error.
+// xattrKey can be empty for listxattr operation.
+type XAttrErrorHandler func(dst, src, xattrKey string, err error) error
+
+type copyDirOpts struct {
+	xeh XAttrErrorHandler
+}
+
+type CopyDirOpt func(*copyDirOpts) error
+
+func WithXAttrErrorHandler(xeh XAttrErrorHandler) CopyDirOpt {
+	return func(o *copyDirOpts) error {
+		o.xeh = xeh
+		return nil
+	}
+}
+
+// WithAllowXAttrErrors allows ignoring xattr errors.
+// Ignored errors are logged with logrus.Warnf.
```

Comment:
is warnf the right log level.. seems this will generate a lot of warnings... maybe debug?

Reply:
Agreed.

Reply:
Maybe s/Warnf/Debugf/g, so that the warnings won't be printed unless debug mode is enabled.

Reply:
I usually hate libs that log because there isn't much you can do about it

Reply:
Removed logrus.

Callers can still provide custom XAttrErrorHandler if logging is needed.